presto-orc/src/main/java/com/facebook/presto/orc/ColumnGroupReader.java

```diff
@@ -98,7 +98,9 @@ public ColumnGroupReader(
             }
             StreamReader streamReader = streamReaders[columnIndex];
             streamReader.setFilterAndChannel(filter, internalChannel, columnIndex, types.get(i));
-            channelToStreamReader.put(internalChannel, streamReader);
+            if (internalChannel != -1) {
```

Comment:
This entire section would be easier to read and more obvious if it were structured like this:

```java
        for (int internalChannel = 0; internalChannel < channelColumns.length; internalChannel++) {
            int columnIndex = channelColumns[internalChannel];
            if (presentColumns != null && !presentColumns.contains(columnIndex)) {
                continue;
            }
            Filter filter = filters.get(columnIndex);
            StreamReader streamReader = streamReaders[columnIndex];
            if (internalChannel < internalChannels.length) {
                streamReader.setFilterAndChannel(filter, internalChannel, columnIndex, types.get(internalChannel));
                channelToStreamReader.put(internalChannel, streamReader);
            }
            else {
                checkArgument(filter != null && streamReaders[columnIndex] == null, "There should be no StreamReader if there is no filter and the value of the column is unreferenced");
                streamReader.setFilterAndChannel(filter, -1, columnIndex, types.get(internalChannel));
            }
        }
```